### DISCUSSION TITLE
Is it possible to scope an atomWithDefault?

### DISCUSSION THREAD

**Discussion by scamden** [pmndrs/jotai#2415] [General]
It seems that adding this to the ScopeProvider atoms list does nothing and I assume that's because it's a derived atom. Is there a way to get access to it's underlying primitive atoms to scope them? Thanks! I'm an enormous jotai fan at this point! 👻

**Comment by scamden** [comment]
fwiw my current working answer was to make a similar version like this

``` ts
export function primitiveAtomWithDefault<Value>(
  baseAtom: WritableAtom<Value | null, [Value | null], void>,
  getDefault: (get: JotaiGetter) => Value
): WritableAtom<Value, [Value], void> {
  return atom(
    (get) => get(baseAtom) ?? getDefault(get),
    (get, set, value: Value) => set(baseAtom, value)
  );
}
```

This way I'm forced to make the primitive atom first and can scope it. (I wrote a lint rule to force every primitive atom in a certain directory to be scoped. Maybe I'm pushing the limits of what jotai-scope is for and should think about bunshi but it seems to be working)

**Comment by dai-shi** [comment]
Is it about [jotai-scope](https://github.com/jotaijs/jotai-scope)?

**Reply by scamden** [reply]
Yes! Sorry should I open there?

**Reply by dai-shi** [reply]
It's fine here, but let's mention @yf-yang who is the main maintainer.
